recurring_contract/models/recurring_contract.py

```diff
@@ -514,17 +521,18 @@ def _updt_invoices_rc(self, vals):
 
         :param vals: the values that are being updated on the contract
         """
-        if any(key in vals for key in ("group_id", "contract_line_ids", "birthday_invoice", "christmas_invoice")):
-            data_invs = {}
-            for contract in self:
-                for inv in contract.mapped("invoice_line_ids.move_id").filtered(
-                        lambda m: m.payment_state == "not_paid" and m.state != "cancel"):
-                    data_invs.update(
-                        inv._build_invoice_data(
-                            contract=contract,
-                            ref=contract.group_id.ref,
-                            pay_mode_id=contract.group_id.payment_mode_id
-                        )
-                    )
-            if data_invs:
-                self.mapped("invoice_line_ids.move_id").update_open_invoices(data_invs)
+        data_invs = {}
+        if "contract_line_ids" in vals:
+            data_invs = self.mapped("open_invoice_ids")._build_invoice_data(contracts=self)
+        if "group_id" in vals:
+            group = self.mapped("group_id")
+            group.ensure_one()
+            group_data = self.mapped("open_invoice_ids")._build_invoice_data(
+                ref=group.ref, pay_mode_id=group.payment_mode_id)
+            for inv_name, inv_data in group_data.items():
+                if inv_name in data_invs:
+                    data_invs[inv_name].update(inv_data)
+                else:
+                    data_invs[inv_name] = inv_data
```

Comment:
```suggestion
                data_invs[inv_name] = inv_data
```
We could simplify this code by writing multiple times the same invoices in the dictionnary for updates. Although that would call multiple times postgresql for the same object.

Reply:
No, a dictionary cannot contain multiple times the same key. Writing inside without checking would override the previous values.